### PR TITLE
Improve LionWeb-specific JSON diff

### PR DESCRIPTION
**Pull request by joswarmer**
Improved diff, used in lionweb-repository

### CODE REVIEW

File: packages/test/src/validation/diff.test.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        if (jsonModel === null || jsonModel2 === null) {
+            console.error("Cannot read model files")
+            return
+        }
+        const diff1 = new LionWebJsonDiff()
+        diff1.diffLwChunk(jsonModel, jsonModel)
+        // No errors expected
+        deepEqual(diff1.diffResult.changes, [])
+
+        // const diff = new LionWebJsonDiff()
```
Comment (dslmeinte) [inline packages/test/src/validation/diff.test.ts:32]: Do we want to keep this commented-out code, or can we simply remove it?

File: packages/validation/src/diff/changes/Change.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import { JsonContext } from "../../issues/index.js"
+
+export type ChangeType =
```
Comment (dslmeinte) [inline packages/validation/src/diff/changes/Change.ts:3]: This starts to feel like a whole separate package. Let's not do that right now, but probably soon-ish, also to aid any ∂ protocol work.

File: release.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env sh
+
+# release all packages
+
+cd packages
+
+for package in "core" "utilities" "cli" "validation" 
+do
+  (cd $package; npm run release)
```
Comment (dslmeinte) [inline release.sh:9]: I'm not sure that `npm run release` is clearer than `npm publish` (to which all `release` scripts are equivalent).

File: packages/validation/package.json [hunk 4]
```diff
@@ -19,7 +19,11 @@
     "testInvalid": "node dist/runners/RunCheckFolder.js ../../../lionweb-integration-testing/testset/invalid",
     "testInvalidWithLanguage": "node dist/runners/RunCheckFolderWithLanguage.js ../../../lionweb-integration-testing/testset/withLanguage/invalid/ ../../../lionweb-integration-testing/testset/withLanguage/myLang.language.json",
     "testValidWithLanguage": "node dist/runners/RunCheckFolderWithLanguage.js ../../../lionweb-integration-testing/testset/withLanguage/valid/ ../../../lionweb-integration-testing/testset/withLanguage/myLang.language.json",
-    "test": "mocha"
+    "test": "mocha",
+    "release": "npm publish",
+    "publish-local": "npm publish --registry http://localhost:4873",
```
Comment (dslmeinte) [inline packages/validation/package.json:24]: What are these “local” scripts for? If we expose it on `main` then we should at least document it.

File: packages/core/src/serialization.ts [hunk 5]
```diff
@@ -35,11 +35,13 @@ export type SerializedReference = {
  * Type definition for an AST node serialized to JSON.
  */
 export type SerializedNode = {
+    // TODO should be Id
     id: string
     classifier: MetaPointer
     properties: SerializedProperty[]
     containments: SerializedContainment[]
     references: SerializedReference[]
+    // TODO inconsistent with the use of Ids
```
Comment (dslmeinte) [inline packages/core/src/serialization.ts:44]: You mean: for consistency, we should fix the 1st TODO, and then both are?

**Review by dslmeinte — APPROVED** [review]

File: packages/test/src/validation/diff.test.ts (hunk 1 above)
Comment (joswarmer) [reply packages/test/src/validation/diff.test.ts:32]: Can be removed.

File: packages/validation/src/diff/changes/Change.ts (hunk 2 above)
Comment (joswarmer) [reply packages/validation/src/diff/changes/Change.ts:3]: Yes, I wat thinking the same.

File: release.sh (hunk 3 above)
Comment (joswarmer) [reply release.sh:9]: Think you are right.

File: packages/validation/package.json (hunk 4 above)
Comment (joswarmer) [reply packages/validation/package.json:24]: I use them to do a local `npm publish` with a local npm registry (in my case `verdaccio`). That way I can use an unpublished version in other projects, which is sometimes very handy. We should document this indeed.

File: packages/core/src/serialization.ts (hunk 5 above)
Comment (joswarmer) [reply packages/core/src/serialization.ts:44]: Inconsistency is between using `Id[]` and `Ids` and `string`. I personally would stick with `Id` and `Id[]`, we use `[]` for other array types (like `SerializedProperty `, etc. ) as well. `Ids` is just obfuscating things.  
So we should change the `string` in the first TODO into `Id` and replace `Ids` (line 21) with `Id[]`.

File: release.sh (hunk 3 above)
Comment (joswarmer) [reply release.sh]: changed

File: packages/core/src/serialization.ts (hunk 5 above)
Comment (joswarmer) [reply packages/core/src/serialization.ts]: fixed

File: packages/validation/package.json (hunk 4 above)
Comment (joswarmer) [reply packages/validation/package.json:23]: documented

File: packages/validation/src/diff/changes/Change.ts (hunk 2 above)
Comment (joswarmer) [reply packages/validation/src/diff/changes/Change.ts:3]: Later ...

File: packages/test/src/validation/diff.test.ts (hunk 1 above)
Comment (joswarmer) [reply packages/test/src/validation/diff.test.ts]: Removed

**Review by joswarmer — COMMENTED** [review]
All done

**Pull request merged**